### PR TITLE
Add a new hardware layout for z70ultra!

### PR DESCRIPTION
**Pull request by mj-bomb**
## Types of Changes

- [ ] Core
- [x] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/melgeek/z70ultra/rev1/rev1.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "z70ultra.h"
+
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/melgeek/z70ultra/rev1/rev1.h:21]:
```suggestion
```

File: keyboards/melgeek/z70ultra/rev1/rules.mk [hunk 2]
```diff
@@ -13,7 +13,9 @@ MOUSEKEY_ENABLE = yes          # Mouse keys
 EXTRAKEY_ENABLE = yes          # Audio control and System control
 CONSOLE_ENABLE = no            # Console for debug
 COMMAND_ENABLE = no            # Commands for debug and configuration
-NKRO_ENABLE = yes           # Enable N-Key Rollover
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
```
Comment (drashna) [inline keyboards/melgeek/z70ultra/rev1/rules.mk:16]: Please revert this change

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/z70ultra/rev1/rules.mk (hunk 2 above)
Comment (mj-bomb) [reply keyboards/melgeek/z70ultra/rev1/rules.mk:16]: The change probably happened unintentionally, does it affect the project?

File: keyboards/melgeek/z70ultra/rev1/rules.mk [hunk 3]
```diff
@@ -13,7 +13,9 @@ MOUSEKEY_ENABLE = yes          # Mouse keys
...
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes              # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/melgeek/z70ultra/rev1/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover```

File: keyboards/melgeek/z70ultra/rev2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes          # Mouse keys
+EXTRAKEY_ENABLE = yes          # Audio control and System control
+CONSOLE_ENABLE = no            # Console for debug
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes              # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/melgeek/z70ultra/rev2/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/z70ultra/config.h [hunk 5]
```diff
@@ -18,19 +18,20 @@
...
 
 /* COL2ROW, ROW2COL*/
 #define DIODE_DIRECTION COL2ROW
 
 /* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
 #define DEBOUNCE 3
 
+/* disable these deprecated features by default */
+#define _CHIBIOS_RT_CONF_VER_6_1_
+#define _CHIBIOS_HAL_CONF_VER_7_1_
```
Comment (drashna) [inline keyboards/melgeek/z70ultra/config.h:31]: I'm not sure that these are okay, here.

File: keyboards/melgeek/z70ultra/rev1/config.h [hunk 6]
```diff
@@ -16,6 +16,13 @@
 
 #pragma once
 
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xEDED
+#define PRODUCT_ID      0x6570
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    MelGeek
+#define PRODUCT         Z70Ultra
+
```
Comment (drashna) [inline keyboards/melgeek/z70ultra/rev1/config.h:25]: This info should be moved to the info.json file.

File: keyboards/melgeek/z70ultra/rev2/config.h [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xEDED
+#define PRODUCT_ID      0x6575
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    MelGeek
+#define PRODUCT         Z70Ultra
```
Comment (drashna) [inline keyboards/melgeek/z70ultra/rev2/config.h:24]: Should be moved to info.json's usb section.

File: keyboards/melgeek/z70ultra/config.h (hunk 5 above)
Comment (mj-bomb) [reply keyboards/melgeek/z70ultra/config.h]: Some documents were missing in the previous project, so I used them. It doesn't matter whether they are there or not

File: keyboards/melgeek/z70ultra/rev2/config.h (hunk 7 above)
Comment (mj-bomb) [reply keyboards/melgeek/z70ultra/rev2/config.h:24]: I seriously tested, after the modification can not compile successfully

Comment (drashna) [reply keyboards/melgeek/z70ultra/rev2/config.h:24]: It does, but it's not just drag and drop. 

It should look something like this:
https://github.com/qmk/qmk_firmware/blob/624ab64750ed05e6da46ce3c261e97c7e39c6206/keyboards/boardsource/3x4/info.json#L17-L21
